### PR TITLE
Adding p5 sketch plugin

### PR DESCRIPTION
**Pull request by nelliemckesson**
Created a plugin to insert the p5 sketch iframes and remove the old figure elements, for the HTML build only. Also updated the magicbook.json file to call the plugin.

### CODE REVIEW

File: plugins/addP5.js [hunk 1]
```diff
@@ -0,0 +1,43 @@
...
+
+  addFrames: function(config, stream, extras, cb) {
+    if(config.format == "html") {
+
+    stream = stream.pipe(through.obj(function(file, enc, cb) {
+        if(!file.$el) file.$el = cheerio.load(file.contents.toString());
+          // Loop through all figures to replace with iframes
+          file.$el('figure[data-pde-sketch^="http"]').each(function(i, el) { 
+            var jel = file.$el(this);
+            var source = jel.attr('data-pde-sketch');
```
Comment (shiffman) [inline plugins/addP5.js]: Let's change this to `data-p5-sketch`! Or just `p5-sketch`? "pde" is a reference to Processing IDE which isn't relevant here so p5 will be more clear!

**Comment by nelliemckesson** [comment]
Alright, I went ahead and made that change, and also added support for the two-col and three-col classes (iframes will get shrunk down and appear inline, copying the current handling), and accompanying captions (e.g., https://github.com/shiffman/The-Nature-of-Code/blob/master/chapters/03_oscillation.html#L834).

There's a corresponding pull request coming on the natureofcode.com repo with some more notes!

**Comment by nelliemckesson** [comment]
(And yeah! Go ahead and pick a chapter and start testing and building and previewing, etc!)

**Comment by vagabesz** [comment]
Bug bit v15&&4

Daniel Shiffman <notifications@github.com> ezt írta (2018. augusztus 9.,
csütörtök):

**Pull request merged**